server/src/main/resources/api-docs.yaml

```diff
@@ -138,6 +152,8 @@ definitions:
       - LIVE
       - DESYNCHRONIZED
       - REMOTE_NOT_FOUND
+      - REMOTE_CHANGED
+      - LOCAL_CHANGED
```

Comment:
I don't think this case can be reliably detected in the current design. I would recommend removing (and just returning the more general "DESYNCHRONIZED" here).

Reply:
wouldn't it be useful to tell client which case it is when we can?

Reply:
Theoretically yes, though practically speaking I think it's probably a minor value add (at most probably a different error message in the MVP).

Keeping metadata state around on disk and trying to make decisions about this sounds like a non-trivial deviation from the original proposed design. Adding a section to the design doc describing how this works might be an easier way to continue this discussion.

Reply:
We make decision around generation regardless. The only deviation is the same data will be persisted on disk the same way storagelinks get persisted on disk